test/scaling/withextrap.jl

```diff
@@ -0,0 +1,40 @@
+
+module ScalingWithExtrapTests
```

Comment:
It wasn't completely for free - I had to add functions `bounds`, `lbound` and `ubound` to make this work - but the composability was one major argument for this design in the first place. I'm glad you like it :)